source/grid/grid_in.cc

```diff
@@ -357,22 +337,20 @@ void GridIn<dim, spacedim>::read_vtk(std::istream &in)
 
           if (dim == 3)
             {
-              for (unsigned int i = 0; i < no_quads; i++) //assigning IDs to bounds.
+              for (unsigned int i = 0; i < n_quads; i++) //assigning IDs to bounds.
                 {
                   double id;
                   in >> id;
-                  Assert (quad_indices[i] < subcelldata.boundary_quads.size(), ExcInternalError());
```

Comment:
I had added this assertion in the previous patch (#3591) but it really isn't necessary any more because we now have a loop over the number of quads and we directly (rather than indirectly as before) index into the array. It's relevant to know that `n_quads == subcelldata.boundary_quads.size()`, so the access in the following line (now 344) is by definition safe.

It may be possible to get rid of the `n_quads` variable altogether and just use `subcelldata.boundary_quads.size()`, which would make this more obvious. I haven't tried that.